apps/builder/app/builder/features/props-panel/controls/combined.tsx

```diff
@@ -0,0 +1,77 @@
+import { TextControl } from "./text";
+import { ColorControl } from "./color";
+import { NumberControl } from "./number";
+import { CheckControl } from "./check";
+import { RadioControl } from "./radio";
+import { SelectControl } from "./select";
+import { BooleanControl } from "./boolean";
+import { FileImageControl } from "./file-image";
+import type { ControlProps } from "../shared";
+
+export const CombinedControl = ({
```

Comment:
does this need to be a component or just a function that renders a component?

Reply:
What is the difference? Is there an issue with it being a component?

Reply:
Component goes through react render circle, its way slower than just a function, do it only if you actually need a component

Reply:
By that logic we should rewrite all our components that don't use hooks as functions. Sounds crazy :)

Reply:
depends, if you need component in the component tree for debugging purposes or because it needs a state - component is the way to go, but when its just a simple render function that doesn't need to be in the tree and will only make the tree less readable - why making it component

Reply:
https://discord.com/channels/955905230107738152/1077912355087863919